`src/time/time_utils.h` unconditionally uses `EOVERFLOW`

Including `time.h` entrypoints such as `asctime`, `gmtime` or `mktime` results in a compile error:
```
In file included from /usr/local/google/home/phosek/llvm/llvm-project/libc/src/time/asctime.cpp:11:
/usr/local/google/home/phosek/llvm/llvm-project/libc/src/time/time_utils.h:94:16: error: use of undeclared identifier 'EOVERFLOW'                                                                                    
   94 |   libc_errno = EOVERFLOW;                                                                                                                                                                                    
      |                ^                                                                                  
1 error generated.
```
This is because `src/time/time_utils.h` [unconditionally uses `EOVERFLOW`](https://github.com/llvm/llvm-project/blob/89b7b3b9952210fbd9bd0db95385bfed69ffc7a3/libc/src/time/time_utils.h#L94) which is only defined in POSIX and not in C standard, but we're trying to avoid including POSIX in baremetal environments. More generally, I don't think that functions defined by the C standard should have unconditional dependencies on POSIX.

CC @michaelrj-google

--- a/libc/config/baremetal/arm/entrypoints.txt
+++ b/libc/config/baremetal/arm/entrypoints.txt
@@ -201,7 +201,12 @@ set(TARGET_LIBC_ENTRYPOINTS
     libc.src.stdlib.strtoull
 
     # time.h entrypoints
+    libc.src.time.asctime
+    libc.src.time.asctime_r
     libc.src.time.difftime
+    libc.src.time.gmtime
+    libc.src.time.gmtime_r
+    libc.src.time.mktime
 
     # internal entrypoints
     libc.startup.baremetal.init

--- a/libc/config/baremetal/riscv/entrypoints.txt
+++ b/libc/config/baremetal/riscv/entrypoints.txt
@@ -197,7 +197,12 @@ set(TARGET_LIBC_ENTRYPOINTS
     libc.src.stdlib.strtoull
 
     # time.h entrypoints
+    libc.src.time.asctime
+    libc.src.time.asctime_r
     libc.src.time.difftime
+    libc.src.time.gmtime
+    libc.src.time.gmtime_r
+    libc.src.time.mktime
 
     # internal entrypoints
     libc.startup.baremetal.init

--- a/libc/src/time/time_utils.h
+++ b/libc/src/time/time_utils.h
@@ -92,7 +92,12 @@ extern int64_t update_from_seconds(int64_t total_seconds, struct tm *tm);
 
 // POSIX.1-2017 requires this.
 LIBC_INLINE time_t out_of_range() {
+#ifdef EOVERFLOW
+  // For non-POSIX uses of the standard C time functions, where EOVERFLOW is
+  // not defined, it's OK not to set errno at all. The plain C standard doesn't
+  // require it.
   libc_errno = EOVERFLOW;
+#endif
   return TimeConstants::OUT_OF_RANGE_RETURN_VALUE;
 }
 

